docs/contributing/docs-and-website-components.md

```diff
@@ -0,0 +1,409 @@
+---
+title: Big News!
+author: Jay Gatsby
+---
+
+<!-- highlight-start -->
+<Pullquote citation="Winston Churchill">
+  To improve is to change, so to be perfect is to have changed often.
+</Pullquote>
+<!-- highlight-end -->
+```
+
+#### Sample
+
+Rendered, the component looks like this:
+
+<Pullquote citation="Winston Churchill">
+  To improve is to change, so to be perfect is to have changed often.
+</Pullquote>
+
+---
+
+## Other available components
+
+Other less commonly used components aren't globally available, but can imported at the top of a markdown file and used in other docs.
+
+---
+
+### Layer model
+
+The `<LayerModel />` was made to help explain concepts of how Gatsby works at a high level. It conceptually breaks Gatsby into different layers and shows how data is pulled, aggregated, and eventually rendered as an app. It's used on pages to help explain and visual how Gatsby works at different levels.
+
+#### Usage
+
+The component takes one prop:
+
+- `initialLayer` - defaults to `Content`, can be one of `Content`, `Build`, `Data`, `View`, `App` that correspond to the different layers
+
+```markdown
+---
+title: GraphQL Concepts in Gatsby
+---
+
+import LayerModel from "../../www/src/components/layer-model"
+
+To help understand how GraphQL works in Gatsby...
+
+<LayerModel initialLayer="Data" />
+```
+
+#### Sample
+
+When used, it looks like this:
+
+<LayerModel initialLayer="Data" />
+
+### Horizontal navigation list
+
+The `<HorizontalNavList />` was made for the [Glossary](/docs/glossary/), and renders a list of links to subheadings on the page in a horizontal format. Because of it's specific use case, it isn't made globally available and can be imported if it's needed on other pages.
+
+#### Usage
+
+It takes two props:
+
+- `slug` - which is provided in the props of the page by default
+- `items` - an array of strings for items to render and wrap with a `<Link />` to subheadings
+
+<!-- prettier-ignore -->
+```markdown
+---
+title: Glossary
+---
+
+import HorizontalNavList from "../../www/src/components/horizontal-nav-list.js"
+
+The glossary defines key vocabulary...
+
+---
+
+<HorizontalNavList
+  slug={props.slug}
+  items={["guide-list", "egghead-embed", "pull-quote", "layer-model", "horizontal-navigation-list"]}
+/>
+```
+
+#### Sample
+
+Rendered, it looks like this:
+
+<HorizontalNavList
+items={[
+"guide-list",
+"egghead-embed",
+"pull-quote",
+"layer-model",
+"horizontal-navigation-list",
+]}
+slug={props.slug}
+/>
+
+### Breadcrumb
+
+The `<Breadcrumb />` component is used in layout files to display the hierarchy of pages a user is currently browsing on.
+
+#### Usage
+
+It takes one prop:
+
+- `location` - an object provided in the props of page templates by default
+- `itemList` - an object comprised of the docs hierarchical structure
+
+<!-- prettier-ignore -->
+```javascript
+import Breadcrumb from "../../www/src/components/docs-breadcrumb"
+
+<Breadcrumb location={props.location} itemList={itemList} />
+```
```

Comment:
We also need to document the `breadcrumbTitle` that goes into sidebar links files.